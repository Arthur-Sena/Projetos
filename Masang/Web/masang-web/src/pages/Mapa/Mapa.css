:root {
    --background: #f2f2f2;
    --laranja-claro: #f28123;
    --vermelho: #B90404;
    --text-color: #868686;
  }

 /* Div Pai  */
.Mapa{
    max-height: 100vh;
    max-width: 100vw;
    margin: 0;
    padding: 0;
    box-sizing: border-box;

    background-color: var(--background);
}

/* Barra de Rolamento */
.DivMapa::-webkit-scrollbar{
    width: 8px;
}
/* 
.DivMapa::-webkit-scrollbar-button{
    background-color: green;
} */

.DivMapa::-webkit-scrollbar-thumb         
{
    background-color: rgba(136,136,136,0.4);
}
/* --------------------- */
/* Div da busca de hemocentro */
.divHemocentro{
    height: 100vh;
    width: 25vw;
    display: flex;
    flex-direction: column;
    
    border-radius: 20;
}

.buscarHemocentro{
    background-color: var(--background);
    display: flex;
    flex-direction: column;
}
/* Input de busca */
.inputFlex{
    display: flex;
    flex-direction: row;
}

.inputBuscar{
    margin-left:3px;
    margin-top: 1.15em;
    height: 2em;
    width: 20vw;
    outline: none;
    border: none;
    background-color:rgba(0,0,0,0);
    font-size: larger;
    font-weight: bolder;

    color: var(--text-color);
}

.inputBottom{
    height: 1vh;
    width: 22vw;
    margin-left: 10px;
    background-image: linear-gradient(to right , var(--laranja-claro), var(--vermelho));

}

/* Lista de Busca */
.infoHemocentro{
    margin-left: 1vw;
    color: var(--text-color);
    margin-right: 10px;

}

.bottomBorder{
    height: 3px;
    width: 19vw;
    background-image: linear-gradient(to right , var(--laranja-claro), var(--vermelho));
    /* margin-right: 2.5vw; */
    margin-right: 15%;
    /* margin-bottom: 2px; */
}

.DivMapa{
    direction: rtl;
    text-align: left;

    overflow: auto;
    margin-left: 10px;
    
    background-color: var(--background);
    /* background-color: var(--background); */
}

.DivMapa > p{
    margin-left: 8px;
}

.logoHome{
    display:flex;
    justify-content: space-around;
    margin-top: auto;
}

/* Div do mapa */
.boxMap {
    width: 75vw;
    position: relative;
    /* margin-left: -2vw; */
}
