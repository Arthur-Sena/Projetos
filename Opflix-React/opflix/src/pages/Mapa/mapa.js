import React, { Component } from "react";
import { Accordion, Card, Button, Table, ButtonToolbar } from 'react-bootstrap';
import Menu from '../../componentes/Menu.js';
import { Map, GoogleApiWrapper, Marker,InfoWindow } from 'google-maps-react';
// import Marker from 'google-map-react';
// import Marker from '../../assets/img/marker.png';

const AnyReactComponent = ({ text }) => <div>{text}</div>;

const PinMarker = ({ Key }) =>
    <div className="pin">
        <img scr={Marker} alt="MArker" className="Marker" />
        <p className="Titulo do Lancamento">{Key}</p>
    </div>


class Mapa extends Component {

    constructor(props) {
        super(props);
        this.state = {
            local: [],
            centro: {
                lat: -23.4549645,
                lng: -46.6960216
            },
        }
    }

    listarLocalizacao = () => {

        fetch('http://192.168.6.115:5000/api/localization')
            .then(response => response.json())
            .then(data => this.setState({ local: data }))

    }

    componentDidMount() {
        console.log(this.state.local)
        this.listarLocalizacao();
    }
    render() {
        return (
            <div style={{ backgroundColor: "#1C1C1C" }}>

                <Menu />
                {/* <br></br>
                <br></br>
                <br></br> */}
                {/* <br></br> */}
                {/* <div style={{ height: '82vh', marginLeft: "8%", width:"84%", border: "solid", borderColor: "Grey" }}> */}


                <Map
                    google={this.props.google}
                    zoom={4}
                    initialCenter={{ lat: -23.4549645, lng: -46.6960216 }}

                // style={{height: '90%',marginBottom:"-50%" }}

                >
                    {this.state.local.map(element => {
                        return (
                            <Marker label={element.nomeLancamento} color="blue" position={{ lat: element.latitude, lng: element.longitude }} />
                            
                        );
                    })}


                </Map>
                {/* </div> */}
                {/* <br></br>
                <br></br>
                <br></br>
                <div style={{ marginLeft: "8%",width:"84%"}}>
                    <Table striped bordered hover variant="dark">
                        <thead>
                            <tr>
                                <th>Filme</th>
                                <th>Latitude</th>
                                <th>Longitude</th>
                            </tr>
                        </thead>
                        <tbody>
                            {this.state.local.map(element => {
                                return (
                                    <tr key={element.id}>
                                        <td>{element.nomeLancamento}</td>
                                        <td>{element.latitude}</td>
                                        <td>{element.longitude}</td>
                                    </tr>
                                );
                            })}
                        </tbody>
                    </Table>
                </div>
                <br></br>
                <br></br>
                <br></br> */}
            </div>
        );
    }
}

export default GoogleApiWrapper({

})(Mapa);