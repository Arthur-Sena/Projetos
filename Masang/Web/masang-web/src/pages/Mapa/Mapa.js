import React, { Component } from 'react';
import { Route, Link, BrowserRouter as Router, Switch, Redirect } from "react-router-dom";

import { Map, GoogleApiWrapper, Marker, InfoWindow } from 'google-maps-react';

//Css + images
import './Mapa.css';
import Lupa from '../../assets/icons/iconBusca.png';
import Logo from '../../assets/logoMasangofc1.png';

import IconMarker from '../../assets/icons/iconfinder1.png';

export class Mapa extends Component {

  constructor(props) {
    super(props);
    this.state = {
      listaHemocentros: [],
      hemocentros: [],
      longitude: "-46.6960216",
      latitude: "-22.4549645",
      buscar: ""
    }
  }

  localizarHemocentro = () => {

    fetch('https://masang.dev.objects.universum.blue/hemocentros')
      .then(response => response.json())
      .then(data => this.setState({ hemocentros: data }) + this.setState({ listaHemocentros: data }))
  }

  atualizaEstado(event) {
    this.setState({ buscar: event.target.value }, () => {
      this.FiltrarHemocentro();
    });
  }

  FiltrarHemocentro() {
    let listaFiltrada = this.state.listaHemocentros;

    if (this.state.buscar !== null && this.state.buscar !== "") {
      listaFiltrada = listaFiltrada.filter(
        x =>
          x.nome.toLowerCase().includes(this.state.buscar.toLowerCase()) ||
          x.endereco.toLowerCase().includes(this.state.buscar.toLowerCase())
      );
    }

    this.setState({ hemocentros: listaFiltrada });

    
  }
      mudarPagina() {
        this.props.history.push('/Hemocentro')
      }

  componentDidMount() {
    this.localizarHemocentro();
  }

  render() {
    return (
      <div className="Mapa"
        style={{ display: 'flex', flexDirection: 'row' }}>

        <div className="divHemocentro">

          <div className="buscarHemocentro">
            {/* Div do Input buscar hemocentro */}
            <div className="inputFlex">
              <img src={Lupa} style={{ height: "1.5em", marginTop: "1.8em", marginLeft: '10px' }}></img>
              <input
                title="a"
                className="inputBuscar"
                placeholder="Buscar Hemocentro"
                value={this.state.buscar}
                onChange={this.atualizaEstado.bind(this)}
              ></input>
            </div>

            <div className="inputBottom"></div>
          </div>

          {/* Div da lista de hemocentros */}
          <article className='DivMapa'>
            {this.state.hemocentros.map(element => {
              let x = JSON.parse(element.endereco);
              return (
                <div className="infoHemocentro">
                  {/* <Link to="/Hemocentro" style={{ textDecoration: "none", color: "#868686" }}  onClick={ () => localStorage.setItem("Xid-HX", element.id) + this.mudarPagina()}>
                    <p  style={{ fontSize: 'large', fontWeight: 'bold' }}>
                      {element.nome}</p>
                  </Link> */}
                  
                  {/* <Link style={{ textDecoration:"none" ,fontSize: '20px', color:'#868686', fontWeight: 'bold', marginTop:'10px' }}>{element.nome}</Link> */}
                  {/* <Link  to="/Hemocentro" style={{textDecoration:"none", color:"#868686" }}><p style={{ fontSize: 'large', fontWeight: 'bold' }}>{element.nome}</p></Link> */}
                  <a  href="/Hemocentro" style={{textDecoration:"none", color:"#868686" }}  onClick={ () => localStorage.setItem("Xid-HX", element.id) + this.mudarPagina()} ><p style={{ fontSize: 'large', fontWeight: 'bold' }}>{element.nome}</p></a>

                  <p style={{ marginTop: "-10px" }}>{x.rua} , {x.numero}</p>
                  <p style={{ marginTop: "-10px" }}>{x.cidade} , {x.estado} - {x.cep}</p>

                  <div className="bottomBorder"></div>
                </div>
              );
            })}
          </article>
      
          <div className="logoHome">
            {/* <Link to="/Services"><img src={Logo} style={{ height: '8vh', marginBottom: "3px" }} /></Link> */}
            <a href="/Services"><img src={Logo} style={{ height: '8vh', marginBottom: "3px" }} /></a>

          </div>
        </div>

        {/* Div do mapa */}
        <div className="boxMap" >
          <Map
            google={this.props.google}
            zoom={8}
            initialCenter={{ lat: this.state.latitude, lng: this.state.longitude }}

            style={{ width: '75vw' }}
          >
            {this.state.hemocentros.map(element => {
              let y = JSON.parse(element.localizacao);

              return (
                <Marker
                  // label={element.nome}
                  // labelStyle={{color:'white'}}
                  position={{ lat: y.latitude, lng: y.longitude }}
                  title={element.nome}
                  icon={IconMarker}
                />
              );
            })}
          </Map>
        </div>

      </div>
    );
  }
}

export default GoogleApiWrapper({})(Mapa);