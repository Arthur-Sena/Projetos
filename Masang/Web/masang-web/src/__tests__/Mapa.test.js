
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import axios from 'axios';
import { Mapa } from '../pages/Mapa/Mapa';



//TESTES INSTÂNTANEOS
describe('Map componente/class [MAPSCREEN]', () => {

    let component;

    beforeEach(() => {
        component = shallow(<Mapa />);
    });

    test('deve injetar corretamente um elemento', () => {
        const componenteInput = renderer.create(<input />).toJSON();
        expect(componenteInput).toMatchSnapshot();
    });

    test('deve realizar um snapshot com sucesso', () => {
        expect(component).toMatchSnapshot();
    })

    test('deve executar a class e ciclos de vida sem crashar', () => {
        const classMap = mount(<Mapa />);
        expect(classMap).toMatchSnapshot();
    });

    test('deve iniciar o state do componente corretamente', () => {
        expect(component.state('hemocentros')).toStrictEqual([]);
        expect(component.state('listaHemocentros')).toStrictEqual([]);
        expect(component.state('longitude')).toBe('-46.6960216');
        expect(component.state('latitude')).toBe('-22.4549645');
        expect(component.state('buscar')).toBe("");
    });

});


//TESTES DE COMPONENTES
describe('Google Map Component', () => {
    
    let component;

    beforeEach(() => {
        component = shallow(<Mapa />);
    });
    
    test('deve renderizar o mapa corretamente com suas props definidas', () => {
        const mapComponent = component.find('Map').first();
        expect(mapComponent).toBeTruthy();
        
        const mockMap = {
            google: component.prop('google'),
            zoom: 8,
            initialCenter: {
                lat: component.state('latitude'),
                lng: component.state('longitude')
            }
        }
        
        expect(mapComponent.length).toBe(1);
        expect(mapComponent.props().zoom).toBe(mockMap.zoom);
        expect(mapComponent.prop('initialCenter')).toStrictEqual(mockMap.initialCenter);
        expect(mapComponent.prop('google')).toBe(mockMap.google);
       
        const marker = component.find('Marker').first();
        expect(marker.length).toBe(0)
    });

    // test('deve retornar uma resposta não nula da requisição', () => {

    // });
    
    // test('deve renderizar os marcadores do mapa com props pré definidas', () => {
    //     const mockMarker = {
    //         position: {},
    //         title: 's',
    //         icon: 'a'
    //     }
        
    //     expect(marker.length).toBe(1);


    // });

})


// describe('Barra de busca hemocentro', () => {
    
//     test('deve retornar uma lista de hemocentros a cada onchange do inpur', () => {

//     });
    
//     test('deve criar uma div (infoHemocentro) para cada item do retorno da requisição', () => {

//     });

//     test('deve redirecionar para a página do hemocentro ao clicar num hemocentro', () => {

//     });

// })

