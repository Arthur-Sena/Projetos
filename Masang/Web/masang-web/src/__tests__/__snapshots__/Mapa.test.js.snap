// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map componente/class [MAPSCREEN] deve executar a class e ciclos de vida sem crashar 1`] = `ReactWrapper {}`;

exports[`Map componente/class [MAPSCREEN] deve injetar corretamente um elemento 1`] = `<input />`;

exports[`Map componente/class [MAPSCREEN] deve realizar um snapshot com sucesso 1`] = `ShallowWrapper {}`;
